test/unit/org/apache/cassandra/config/DatabaseDescriptorTest.java

```diff
@@ -644,4 +650,29 @@ public void testDataFileDirectoriesMinTotalSpaceInGB() throws IOException
         fileStoreMultiset.add(mockSmallFileStore);
         assertEquals(0L, DatabaseDescriptor.getDataFileDirectoriesMinTotalSpaceInGB(fileStoreMultiset));
     }
+
+    @Test
+    public void testResetUnsafe()
+    {
+        assertTrue(DatabaseDescriptor.isDaemonInitialized());
```

Comment:
a lot of disaster potential in that test... :)
e.g. it could potentially break other test cases that run after it...

well.. we'll see :)

Reply:
Yes, I was concerned about that and during my testing I added several additional temporary tests all calling `testResetUnsafe` to try and mix up the test ordering.

Doing so, I found two things:
* that `testConfigurationLoader` needed to clear the `cassandra.config.loader` property that was set; otherwise if `testResetUnsafe` runs after this test it would fail.
* that `resetUnsafe` needs to close `DynamicEndpointSnitch`, otherwise it's mbean remains registered.

Reply:
:+1: